projects/core/src/config/index.ts

```diff
@@ -1,7 +1,17 @@
-export * from './config.module';
+export * from './config-initializer/index';
+export * from './config-validator/index';
+export {
+  Config,
+  ConfigChunk,
+  ConfigModule,
+  configurationFactory,
+  DefaultConfigChunk,
+  provideConfig,
+  provideConfigFactory,
+  provideDefaultConfig,
+  provideDefaultConfigFactory,
+} from './config.module';
```

Comment:
explicit export of all parts to avoid exposing `configurationFactoryProvidedInRoot`